lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -43,18 +46,38 @@ export function WithRpcSessions({ children }: { children: React.ReactNode }) {
     return <RpcSessionsContext.Provider value={sessions}>{children}</RpcSessionsContext.Provider>
 }
 
-const fakeRpcSession: RpcSessionAtPos = {
+const noCtxRpcSession: RpcSessionAtPos = {
     call: async () => {
-        throw new Error('no rpc context set')
+        throw new Error('no RPC context set')
+    },
+}
+
+const noPosRpcSession: RpcSessionAtPos = {
+    call: async () => {
+        throw new Error('no position context set')
     },
 }
 
 export function useRpcSessionAtTdpp(pos: TextDocumentPositionParams): RpcSessionAtPos {
-    return React.useContext(RpcSessionsContext)?.connect(pos) || fakeRpcSession
+    return React.useContext(RpcSessionsContext)?.connect(pos) || noCtxRpcSession
 }
 
 export function useRpcSessionAtPos(pos: DocumentPosition): RpcSessionAtPos {
     return useRpcSessionAtTdpp(DocumentPosition.toTdpp(pos))
 }
 
-export const RpcContext = React.createContext<RpcSessionAtPos>(fakeRpcSession)
+/** @deprecated use {@link useRpcSession} instead */
+export const RpcContext = React.createContext<RpcSessionAtPos>(noCtxRpcSession)
```

Comment:
What is the path to fully replacing this with `useRpcSession`? Why is it still necessary to use `RpcContext` in the code base?

Reply:
Existing widgets use the `RpcContext`, so for backwards compatibility we must set it to the correct value. It could potentially be removed later on in the future.

Reply:
Ah, that makes sense. Thanks!